ui/src/shared/parsing/flux/response.ts

```diff
@@ -25,10 +25,53 @@ export const parseResponse = (response: string): FluxTable[] => {
   }
 
   return trimmedResponse.split(/\n\s*\n/).reduce((acc, chunk) => {
+    if (chunk.indexOf('#datatype') === -1) {
+      return [...acc, ...parseTablesWithoutAnnotations(chunk)]
+    }
     return [...acc, ...parseTables(chunk)]
   }, [])
 }
 
+export const parseTablesWithoutAnnotations = (
+  nonAnnotationLines: string
+): FluxTable[] => {
+  if (_.isEmpty(nonAnnotationLines)) {
+    // A response may be truncated on an arbitrary line. This guards against
+    // the case where a response is truncated on annotation data
+    return []
+  }
+
+  const nonAnnotationData = Papa.parse(nonAnnotationLines).data
+  const headerRow = nonAnnotationData[0]
+
+  interface TableGroup {
+    [tableId: string]: string[]
+  }
+
+  // Group rows by their table id
+  const tablesData = Object.values(
+    _.groupBy<TableGroup[]>(nonAnnotationData.slice(1), row => row[''])
```

Comment:
I don't follow what row[''] is referring to? it looks like it makes any string to undefined? el confusedo.